app/forms/prior_authority/document_uploader.rb

```diff
@@ -0,0 +1,74 @@
+module PriorAuthority
+  # This concern expects there to be an attribute called `file_upload` containing the temporary file
+  # and a method called document that returns the appropriate SupportingDocument object
+  # It will handle validation automatically, but you will need to call `save_file` in your
+  # `persist` method to get it to actually upload it and persist the metadata
+  module DocumentUploader
+    extend ActiveSupport::Concern
+
+    included do
+      validate :validate_uploaded_file
+    end
+
+    def document_already_uploaded?
+      document.file_name.present?
+    end
+
+    private
+
+    def save_file
+      return if file_upload.blank?
+
+      file_path = file_uploader.upload(file_upload)
+      save_file_metadata(file_path)
+    end
+
+    def save_file_metadata(file_path)
+      document.update!(
+        document_type: SupportingDocument::QUOTE_DOCUMENT,
+        file_name: file_upload.original_filename,
+        file_type: file_upload.content_type,
+        file_size: file_upload.tempfile.size,
+        file_path: file_path
+      )
+    end
+
+    def validate_uploaded_file
+      if file_upload.nil?
+        return if document_already_uploaded?
+
+        add_file_upload_error(I18n.t('shared.shared_upload_errors.file_not_present', file: 'primary quote'))
+      elsif !file_size_within_limit?
+        add_file_upload_error(I18n.t('shared.shared_upload_errors.file_size_limit', max_size: '20MB'))
+      elsif !supported_filetype?
+        add_file_upload_error(I18n.t('shared.shared_upload_errors.file_type',
+                                     file_types: I18n.t('shared.shared_upload_errors.file_types')))
+      elsif suspected_malware?
+        add_file_upload_error(I18n.t('shared.shared_upload_errors.malware'))
+      end
+    end
+
+    def add_file_upload_error(string)
+      errors.add(:file_upload, string)
+    end
+
+    def file_size_within_limit?
+      file_upload.tempfile.size <= ENV.fetch('MAX_UPLOAD_SIZE_BYTES', nil).to_i
+    end
+
+    def supported_filetype?
+      SupportedFileTypes::QUOTE_DOCUMENT.include? file_upload.content_type
+    end
+
+    def suspected_malware?
+      file_uploader.scan_file(file_upload)
+      false
+    rescue FileUpload::FileUploader::PotentialMalwareError
+      true
```

Comment:
**not a blocker** perhaps we could/should be capturing the StandardError exceptions to sentry here too. possibly even moving the 
rescue blocks in controllers to somewhere in here if feasable
```rb
    rescue StandardError => e
        Sentry.capture_exception(e)
        @form_object.errors.add(:base, t('shared.shared_upload_errors.unable_upload'))
```